### PR TITLE
fix(parser): introduce ParenExpression

### PR DESCRIPTION
**Pull request by affo**
I chose to make the `SourceLocation` returned by `Node.Location()` mutable in order to make it possible to override the source location.

This is needed in order to change the location of a retuned expression once it is inside of parenthesis.

If I try to introduce the change in the concrete expression, then I don't have enough context and I don't know if I am inside of parens.

If I introduce the change at an higher level (`parseParenExpression`), then too many things get the parenthesis in (for example: `f = (a) => ...`, the identifier `a` becomes `(a)`).

That seemed to me a good trade-off.

I also had to make changes to the Rust part and create a method to set the location for an expression.

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

**Comment by affo** [comment]
@jsternberg `ParenExpression` introduced.

This is way better

File: internal/parser/parser.go [hunk 1]
```diff
@@ -1036,8 +1036,8 @@ func (p *parser) parseObjectLiteral() ast.Expression {
 }
 
 func (p *parser) parseParenExpression() ast.Expression {
-	pos, _ := p.open(token.LPAREN, token.RPAREN)
```
Comment (jsternberg) [inline internal/parser/parser.go]: Why the name change?

File: semantic/analyze.go [hunk 2]
```diff
@@ -281,6 +281,9 @@ func analyzeExpression(expr ast.Expression) (Expression, error) {
 		return analyzeIdentifierExpression(expr)
 	case *ast.StringExpression:
 		return analyzeStringExpression(expr)
+	case *ast.ParenExpression:
+		// Unwrap the parenthesis and analyze underlying Expression.
+		return analyzeExpression(expr.Expression)
```
Comment (jsternberg) [inline semantic/analyze.go]: 👍

File: internal/parser/parser.go (hunk 1 above)
Comment (affo) [reply internal/parser/parser.go]: cause I forgot to revert it back, fixing now

**Pull request merged**